src/main/java/com/sonyericsson/jenkins/plugins/bfa/PluginImpl.java

```diff
@@ -629,6 +631,63 @@ public void setSlackFailureCategories(String slackFailureCategories) {
         this.slackFailureCategories = slackFailureCategories;
     }
 
+    /**
+     * Get if causes from the current knowledge base should be included in the JCasC export is enabled.
+     * @return exportCausesToJCasCEnabled - on or off. null == off.
+     */
+    public boolean getExportCausesToJCasCEnabled() {
+        return exportCausesToJCasCEnabled;
+    }
+
+    /**
+     * Set if this feature is enabled or not. When on, causes will in the JCasC export.
+     *
+     * @param exportCausesToJCasCEnabled on or off. null == off.
+     */
+    @DataBoundSetter
+    public void setExportCausesToJCasCEnabled(boolean exportCausesToJCasCEnabled) {
+        this.exportCausesToJCasCEnabled = exportCausesToJCasCEnabled;
+    }
+
+    /**
+     * Set the causes.
+     *
+     * @param causes value
+     */
+    @DataBoundSetter
+    public void setCauses(List<FailureCause> causes) {
+        if (knowledgeBase != null) {
+            for (FailureCause cause:causes) {
+                try {
+                    knowledgeBase.saveCause(cause);
```

Comment:
I am quite certain this will create duplicates for anything else than the file based knowledge base. Unless the `id` field is set. At least the javadoc should reflect that.

Reply:
Writing a unit test for that should be relatively easy. There are a couple of integration tests using an embedded mongodb that you can take inspiration from.

Reply:
I've added a note about duplicates to the javadoc, there isn't much I can do about missing ids short of skipping entries that don't have one, thoughts on the preferred behavior?